nixos/modules/services/cluster/kubernetes/dns.nix

```diff
@@ -48,13 +26,46 @@ in {
       default = "cluster.local";
       type = types.str;
     };
+
+    kube-dns = mkOption {
+      description = "Docker image to seed for the kube-dns main container.";
+      type = types.attrs;
+      default = {
+        imageName = "k8s.gcr.io/k8s-dns-kube-dns-amd64";
```

Comment:
Does k8s run on ARM? Probably out of scope for this PR otherwise.

Reply:
Haven't tested on other archs. At least now you can configure another image íf you want to :) Before this change, the image details were hardcoded.